### PR TITLE
take a db dump prior to migrations on deploy

### PR DESCRIPTION
**Pull request by greatbacon**
This PR adds functionality to take a pg_dump of the db prior to running a migration in prod

### CODE REVIEW

File: src/lib/db/migrate.ts [hunk 1]
```diff
@@ -21,6 +24,12 @@ export const migrate = async (prod: boolean, log: Logger): Promise<void> => {
 		return;
 	}
 
+	if (prod) {
+		log.info('backing up db');
+		await spawn('sudo', ['-i', '-u', 'postgres', 'pg_dump', 'felt', '-f', `${BACKUP_FILE}`]);
```
Comment (ryanatkn) [inline src/lib/db/migrate.ts:29]:
```suggestion
		await spawn('sudo', [
			'-i',
			'-u',
			defaultPostgresOptions.username,
			'pg_dump',
			defaultPostgresOptions.database,
			'-f',
			BACKUP_FILE,
		]);
```

File: src/lib/db/migrate.ts [hunk 2]
```diff
@@ -21,6 +24,12 @@ export const migrate = async (prod: boolean, log: Logger): Promise<void> => {
...
+		await spawn('mv', [`${BACKUP_DIR}/${BACKUP_FILE}`, '.']);
```
Comment (ryanatkn) [inline src/lib/db/migrate.ts:30]:
```suggestion
		await spawn('sudo', ['mv', `${BACKUP_DIR}/${BACKUP_FILE}`, '.']);
```

fails for me locally without this

File: src/lib/db/migrate.ts [hunk 3]
```diff
@@ -21,6 +24,20 @@ export const migrate = async (prod: boolean, log: Logger): Promise<void> => {
...
+		await spawn('sudo', [
+			'-i',
+			'-u',
+			defaultPostgresOptions.username,
+			'pg_dump',
+			defaultPostgresOptions.database,
+			'-f',
+			BACKUP_FILE,
+		]);
+		await spawn('sudo', ['mv', `${BACKUP_DIR}/${BACKUP_FILE}`, '.']);
```
Comment (ryanatkn) [inline src/lib/db/migrate.ts:38]:
```suggestion

```

Edited my other comment, I think we should just remove this, unless there's a need for it?

**Review by ryanatkn — APPROVED** [review]

File: src/lib/db/migrate.ts [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import ley from 'ley';
 import {defaultPostgresOptions} from '$lib/db/postgres.js';
 import type {Logger} from '@feltcoop/util/log.js';
+import {spawn} from '@feltcoop/util/process.js';
 
 import {MIGRATIONS_DIR, MIGRATIONS_DIR_PROD} from '$lib/db/migration';
 
 // Note: this requires the dependency `tsm`
+const BACKUP_DIR = '/var/lib/postgresql';
```
Comment (ryanatkn) [inline src/lib/db/migrate.ts:9]:
```suggestion
// Note: this requires the dependency `tsm`
const BACKUP_DIR = '/var/lib/postgresql';
```

Comment (greatbacon) [reply src/lib/db/migrate.ts:9]: ah, thank you for cleaning up my mess lol

**Pull request merged**